test/e2e/util/util.go

```diff
@@ -727,6 +728,113 @@ func EnsureAllRoutesUseHCPRouter(t *testing.T, ctx context.Context, hostClient c
 	})
 }
 
+func EnsureNetworkPolicies(t *testing.T, ctx context.Context, c crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+	t.Run("EnsureNetworkPolicies", func(t *testing.T) {
+		t.Run("EnsureComponentsHaveNeedManagementKASAccessLabel", func(t *testing.T) {
+			// Check for all components expected to have NeedManagementKASAccessLabel.
+			want := []string{
+				"cluster-network-operator",
+				"ignition-server",
+				"router",
+				"cluster-storage-operator",
+				"cluster-csi-snapshot-controller-operator",
+				"machine-approver",
+				"cluster-autoscaler",
+				"cluster-node-tuning-operator",
+				"capi-provider-controller-manager",
+				"cluster-api",
+				"control-plane-operator",
+				"hosted-cluster-config-operator",
+			}
+			g := NewWithT(t)
+			err := checkDeploymentsHaveLabel(ctx, c, want, hostedCluster.Namespace, client.MatchingLabels{suppconfig.NeedManagementKASAccessLabel: "true"})
+			g.Expect(err).ToNot(HaveOccurred(), err)
+
+		})
+
+		t.Run("EnsureEgressTrafficToManagementKAS", func(t *testing.T) {
+			command := []string{
+				"curl",
+				"--connect-timeout",
+				"2",
+				"-k",
+				// Default KAS advertised address.
+				"https://172.20.0.1:6443",
```

Comment:
where did this IP come from? isn't better to use use the Kubernetes service DNs record `https://kubernetes.default.svc:443`?

Reply:
This is our `DefaultAdvertiseAddress = "172.20.0.1"`.
I want to query the pod network directly to remove chances of Service network resolution hiding the real test.